pandas/core/missing.py

```diff
@@ -542,14 +542,55 @@ def _cubicspline_interpolate(xi, yi, x, axis=0, bc_type="not-a-knot", extrapolat
 
 def interpolate_2d(
     values,
-    method="pad",
-    axis=0,
-    limit=None,
+    method: str = "pad",
+    axis: Axis = 0,
+    limit: Optional[int] = None,
+    limit_area: Optional[str] = None,
 ):
     """
     Perform an actual interpolation of values, values will be make 2-d if
     needed fills inplace, returns the result.
     """
+
+    # `limit_area` is not supported by `pad_2d` and `backfill_2d`. Hence, the
+    # following code block does a recursive call and applies the interpolation
+    # and `limit_area` logic along a certain axis.
+    if limit_area is not None:
+
+        def func(values):
```

Comment:
can you make this a module level function (can be private is ok), give it a real name, with doc-strings & types.

Reply:
Done